Library/Homebrew/env_config.rb

```diff
@@ -239,12 +242,20 @@ module EnvConfig
         default:     "git",
       },
       HOMEBREW_GITHUB_API_TOKEN:                 {
-        description: "Use this personal access token for the GitHub API, for features such as " \
-                     "`brew search`. You can create one at <https://github.com/settings/tokens>. If set, " \
-                     "GitHub will allow you a greater number of API requests. For more information, see: " \
-                     "<https://docs.github.com/en/rest/overview/rate-limits-for-the-rest-api>" \
-                     "\n\n    *Note:* Homebrew doesn't require permissions for any of the scopes, but some " \
-                     "developer commands may require additional permissions.",
+        description: <<~EOS,
+          Use this personal access token for the GitHub API, for features such as `brew search`.
+          You can create one at <https://github.com/settings/tokens>.
+          If set, GitHub will allow you a greater number of API requests.
+          For more information, see: "<https://docs.github.com/en/rest/overview/rate-limits-for-the-rest-api>"
+
+              *Note:* Homebrew doesn't require permissions for any of the scopes,
+                      but some developer commands may require additional permissions.
+
+              *Note:* If set, this token will also be used to authenticating against `ghcr.io` unless
+                      `HOMEBREW_DOCKER_REGISTRY_TOKEN` or `HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN` has been set.
+                      In this case, the token will require the `packages` permission to be set.
+                      `HOMEBREW_GITHUB_PACKAGES_TOKEN` is not used for this permission, but only for uploading packages.
```

Comment:
```suggestion
```

Remove this and move to the `HOMEBREW_GITHUB_PACKAGES_TOKEN` description (if there is one).